### ISSUE TITLE
Trigger Colorscheme Autocommand

### ISSUE THREAD

**Issue by austinliuigi** [vimpostor/vim-lumen#5]
When the colorscheme changes after lumen identifies a change in the system dark mode preference, it doesn't trigger the Colorscheme autocommand. Is this just something to do with vim not triggering an autocommand when the trigger is inside another autocommand or can this possibly be fixed in lumen?

**Comment by vimpostor** [comment]
This works for me, but I am not sure if I understood you correctly.

For example if I set another colorscheme in the Lumen callback and add an autocmd that prints the new colorscheme name:
```vim
au User LumenLight colorscheme shine
au User LumenDark colorscheme industry
au ColorScheme * echom expand('<amatch>')
```

Then it prints the new colorscheme name every time, so the ColorScheme autocmd seems to be working for me.
If that's not what you meant, please provide precise instructions of the problem.

**Comment by austinliuigi** [comment]
So I just tried it with `au ColorScheme * echom expand('<amatch>')` and it works as you showed, printing the name of the colorscheme that triggered the autocommand.

However, if the command is instead setting a custom highlight color, it doesn't seem to work. For example, the following command didn't set the ALEErrorSign highlight group after the colorscheme change:
```
au ColorScheme * highlight! link ALEErrorSign Error
```

After some googling, I found that adding the nested keyword in the lumen autocommands fixes this issue:
```
au User LumenLight nested colorscheme shine
au User LumenDark nested colorscheme industry
```

I'm guessing this has something to do with the order of when the autocommand gets triggered and when the commands execute, but I'm still not entirely sure why it behaves this way.
